src/containers/hierarchy.jsx

```diff
@@ -74,6 +76,44 @@ const filterHierarchy = (text, folder, folders) => {
   return result
 }
 
+
+
+const buildTree = (parents, parent, folders) => {
+  const items = []
+  for (const child of (parents[parent] || [])) {
+    if (!child) continue
+
+    const nchild = {
+      key: child.id,
+      data: {
+        name: child.name,
+        label: child.label || child.name,
+        status: child.status,
+        folderType: child.folderType,
+        hasTasks: child.hasTasks,
+        parents: child.parents,
+        body: (
+          <CellWithIcon
+            icon={folders[child.folderType]?.icon}
+            text={child.label}
+            name={child.name}
+          />
+        ),
+      }
+     
+    }
+    if (child.id in parents) {
+      nchild.children = buildTree(parents, child.id, folders)
```

Comment:
We will probably want to optimise the build tree. Calling the same function inside of itself can get super slow. There are some other tricks we can do to speed it up!